src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php

```diff
@@ -238,11 +246,20 @@ public function send(string $body, array $headers, int $delay = 0): void
             $this->setup();
         }
 
-        $this->call($this->getQueueUrl(), [
+        $messageBody = json_encode(['body' => $body, 'headers' => $headers]);
+
+        $parameters = [
             'Action' => 'SendMessage',
-            'MessageBody' => json_encode(['body' => $body, 'headers' => $headers]),
+            'MessageBody' => $messageBody,
             'DelaySeconds' => $delay,
-        ]);
+        ];
+
+        if ($this->isFifoQueue($this->configuration['queue_name'])) {
+            $parameters['MessageGroupId'] = __METHOD__;
+            $parameters['MessageDeduplicationId'] = md5($messageBody);
```

Comment:
What's about allowing user overriding those value (via a Stamp)?

Reply:
Doing md5($messageBody) will be the same as the default, right?

If no duplication ID is provided it will use the message body as key.

Reply:
Btw, using sha1 over md5 is quicker if I remember correctly.